+adi/+AD7124/Base.m

```diff
@@ -0,0 +1,120 @@
+classdef Base < adi.common.Rx & adi.common.RxTx & ...
+         matlabshared.libiio.base & adi.common.Attribute & ...
+         adi.common.RegisterReadWrite & adi.common.Channel
+    % AD7124 Precision ADC Class
+    % AD7124-4 is a 8 channel ADC
+    % AD7124-8 is a 16 channel ADC
+
+    properties (Nontunable)
+        % SampleRate Sample Rate
+        %   Baseband sampling rate in Hz, specified as a scalar
+        %   in samples per second.
+        SampleRate = '19200'
+
+        % SamplesPerFrame Samples Per Frame
+        %   Number of samples per frame, specified as an even positive
+        %   integer.
+        SamplesPerFrame = 400
+    end
+    
+    % isOutput
+    properties (Hidden, Nontunable, Access = protected)
+        isOutput = false
+    end
+
+    properties (Nontunable, Hidden, Constant)
+        Type = 'Rx'
+    end
+
+    properties (Nontunable, Hidden)
+        Timeout = Inf
+        kernelBuffersCount = 1
+        dataTypeStr = 'int32'
+        phyDevName
+        devName
+    end
+
+    properties (Hidden, Constant)
+        ComplexData = false
+    end
+
+    methods
+    %% Constructor
+        function obj = Base(phydev,dev,varargin)
+            coder.allowpcode('plain');
+            % Initialize the Rx object
+            obj = obj@matlabshared.libiio.base(varargin{:});
+            obj.enableExplicitPolling = false;
+            obj.EnabledChannels = 1;
+            obj.BufferTypeConversionEnable = true;
+            obj.phyDevName = phydev;
+            obj.devName = dev;
+
+            % Check if uri has been specified, else throw an error
```

Comment:
This would work, but if there's not an express need to connect to the device during instantiation, we can drop this (lines 53 - 71). 
We could then modify the example scripts to set the uri _after_ the instantiation.

Reply:
My reasoning here is, there exists a more concise way of parsing input arguments - but in the context of this driver, we don't need to provide the uri at init time, at all. 
So long as they set it before calling the system object ( `data = rx()` ), it shouldn't be an issue.

Reply:
Makes sense. I have made the changes